File: syncloud/src/main/java/org/syncloud/android/AcraLogEmailerFactory.java
package org.syncloud.android;

import android.content.Context;
import androidx.annotation.NonNull;

import org.acra.config.CoreConfiguration;
import org.acra.sender.ReportSender;
import org.acra.sender.ReportSenderFactory;

public class AcraLogEmailerFactory implements ReportSenderFactory {
    public AcraLogEmailerFactory() {

    }

    @Override
    public ReportSender create(@NonNull Context context, @NonNull CoreConfiguration config) {
        return new AcraLogEmailer(context, config);
    }
}

File: syncloud/src/main/java/org/syncloud/android/AcraLogEmailerFactory.kt
package org.syncloud.android

import android.content.Context
import org.acra.sender.ReportSenderFactory
import org.acra.config.CoreConfiguration
import org.acra.sender.ReportSender
import org.syncloud.android.AcraLogEmailer

class AcraLogEmailerFactory : ReportSenderFactory {
    override fun create(context: Context, config: CoreConfiguration): ReportSender =
        AcraLogEmailer(context, config)
}